pkg/gofr/config/godotenv_test.go

```diff
@@ -99,20 +99,23 @@ func Test_EnvFailureWithHyphen(t *testing.T) {
 
 	logger := logging.NewMockLogger(logging.DEBUG)
 
-	err := createConfigsDirectory()
-	if err != nil {
-		t.Error(err)
-	}
+	configFiles := []string{".env", ".local.env"}
 
-	// Call the function to create the .env file
-	createEnvFile(t, ".env", envData)
+	for _, file := range configFiles {
+		err := createConfigsDirectory()
+		if err != nil {
+			t.Error(err)
+		}
 
-	defer os.RemoveAll("configs")
+		createEnvFile(t, file, envData)
 
-	env := NewEnvFile("configs", logger)
+		env := NewEnvFile("configs", logger)
+
+		assert.Equal(t, "test", env.GetOrDefault("KEY-WITH-HYPHEN", "test"), "TEST Failed.\n godotenv failure with hyphen")
+		assert.Equal(t, "", env.Get("UNABLE_TO_LOAD"), "TEST Failed.\n godotenv failure with hyphen")
 
-	assert.Equal(t, "test", env.GetOrDefault("KEY-WITH-HYPHEN", "test"), "TEST Failed.\n godotenv failure with hyphen")
-	assert.Equal(t, "", env.Get("UNABLE_TO_LOAD"), "TEST Failed.\n godotenv failure with hyphen")
+		os.RemoveAll("configs")
```

Comment:
This look like an error to that to me.

https://github.com/gofr-dev/gofr/pull/854#discussion_r1684278442

Reply:
error as in using RemoveAll at end manually instead of defer?

Reply:
Yes, somehow

If the code if refactored later,  and a return is added, then the configs folder and its env file won't be cleaned.

Also, it's a common pattern to create temporary file then to immediately plan their removal, either with a defer, or t.Cleanup when in file